src/css/hiddenVisibleSelectors.js

```diff
@@ -1,15 +1,21 @@
 define( [
 	"../core",
+	"../var/document",
 	"./support",
 	"../selector",
 	"../css"
-], function( jQuery, support ) {
+], function( jQuery, document, support ) {
 
 function getDisplay( elem ) {
 	return elem.style && elem.style.display || jQuery.css( elem, "display" );
 }
 
 function filterHidden( elem ) {
+
+	// Disconnected elements are considered hidden
+	if ( !jQuery.contains( elem.ownerDocument || document, elem ) ) {
```

Comment:
I feel nervous not doing it tho. Seems like whenever I leave it out, I hit a problem eventually.